service/src/main/java/dk/dbc/opensearch/ServiceBean.java

```diff
@@ -0,0 +1,241 @@
+
+    private static final Logger log = LoggerFactory.getLogger(ServiceBean.class);
+
+    @Inject
+    BadgerFishSingle badgerFishSingle;
+
+    @Inject
+    Settings settings;
+
+    @Inject
+    RemoteIPAddress remoteIPAddress;
+
+    @Inject
+    IndexHtmlBean indexHtml;
+
+    @Inject
+    GetObjectProcessorBean getObjectProcessor;
+
+    @Inject
+    InfoProcessorBean infoProcessor;
+
+    @Inject
+    SearchProcessorBean searchProcessor;
+
+    @GET
+    public Response index(@Context UriInfo context, @Context HttpHeaders headers, @Context HttpServletRequest httpRequest) {
+        String peer = remoteIPAddress.ip(headers, httpRequest);
+        try (MDCLog mdc = MDCLog.mdc()
+                .withPeer(peer)) {
+            MultivaluedMap<String, String> params = context.getQueryParameters();
+            if (params.isEmpty()) {
+                return Response.ok(indexHtml.getBytes())
+                        .type(MediaType.TEXT_HTML_TYPE)
+                        .build();
+            } else {
+                return badRequest("Support for uri parameters not implemented yet");
+            }
+        }
+    }
+
+    @GET
+    @Path("{path: .+}")
+    public Response docroot(@PathParam("path") String path, @Context HttpHeaders headers, @Context HttpServletRequest httpRequest) {
+        String peer = remoteIPAddress.ip(headers, httpRequest);
+        try (MDCLog mdc = MDCLog.mdc()
+                .withPeer(peer)) {
+            return ResourceStreamingOutput.docroot(path);
+        }
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    public Response post(@FormParam("req") String req, @Context HttpHeaders headers, @Context HttpServletRequest httpRequest) {
+        try (ByteArrayInputStream bis = new ByteArrayInputStream(req.getBytes(StandardCharsets.UTF_8))) {
+            try (PushbackInputStream is = new PushbackInputStream(bis, 1024)) {
+                is.mark(1024);
+                int c = is.read();
+                while (Character.isWhitespace(0)) {
+                    c = is.read();
+                }
+                if (c == -1)
+                    throw new IOException("Empty request");
+                is.reset();
+                if (c == '{')
+                    return processInputStream(headers, httpRequest, is, RequestParserJSON::new);
+                else
+                    return processInputStream(headers, httpRequest, is, RequestParser::new);
+            }
+        } catch (IOException ex) {
+            log.error("post (form): {}", ex.getMessage());
+            log.debug("post (form): ", ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @POST
+    @Consumes({MediaType.APPLICATION_XML,
+               "application/soap+xml"})
+    @Produces({MediaType.APPLICATION_XML,
+               MediaType.APPLICATION_JSON})
+    public Response postXML(InputStream is, @Context HttpHeaders headers, @Context HttpServletRequest httpRequest) {
```

Comment:
Har du overvejet @Suspended AsyncResponse for at frigive http tråden hurtigt? Noget i stil med

public void postXML(@Suspended final AsyncResponse asyncResponse, ...) {
    CompletableFuture
                .runAsync(() -> processInputStream(...), managedExecutor)
                .thenApply((response) -> asyncResponse.resume(response));
}

Reply:
Det ser ikke ud til at virke med StreamingOutput, den kommer i hvert fald ikke ind i write()